### ISSUE TITLE
Producing incorrect HasField in presence of KindSignature

### ISSUE THREAD

**Issue by mageshb** [ndmitchell/record-dot-preprocessor#28]
Problem happens with Preprocessor.
Following works fine,
```haskell
data User f = User
 { name :: String }
```

Following fail (ignores the type parameter with KindSignature),
```haskell
data UserF (f :: Type -> Type) = UserF
 { name :: String } 
```
Error
```
[1 of 1] Compiling MyLib            ( src/MyLib.hs, interpreted )

src/MyLib.hs:201:29: error:
    • Expecting one more argument to ‘UserF’
      Expected a type, but ‘UserF’ has kind ‘(* -> *) -> *’
    • In the second argument of ‘Z.HasField’, namely ‘(UserF)’
      In the instance declaration for
        ‘Z.HasField "name" (UserF) (String)’
Failed, no modules loaded.

src/MyLib.hs:201:29: error:
    • Expecting one more argument to ‘UserF’
      Expected a type, but ‘UserF’ has kind ‘(* -> *) -> *’
    • In the second argument of ‘Z.HasField’, namely ‘(UserF)’
      In the instance declaration for
        ‘Z.HasField "name" (UserF) (String)’

```

**Comment by ndmitchell** [comment]
Thanks for the report. I've pushed a new version which has a fix and a test for kind signatures. Can you try master and see if that fixes it for you? If it does, or if you are unable to try from master, I'll make a release.

**Comment by mageshb** [comment]
The new version works. Thank you for the quick fix.

**Comment by ndmitchell** [comment]
Thanks for the test. I released 0.2.5 including the fix.
